### PR TITLE
Added functionality to access terminal window height (number of lines)

### PR DESCRIPTION
**Pull request by fastrgv**
For several of my apps I needed the window height.  This change has been tested on OSX, Windows, and several flavors of linux.
See, for example, "RetroArcade". This is one of my projects on GitHub.

### CODE REVIEW

File: src/terminals.c [hunk 1]
```diff
@@ -121,7 +121,9 @@ int gnatcoll_terminal_width(int forStderr) {
       GetStdHandle (forStderr ? STD_ERROR_HANDLE : STD_OUTPUT_HANDLE);
    CONSOLE_SCREEN_BUFFER_INFO csbiInfo;
    if (GetConsoleScreenBufferInfo (handle, &csbiInfo)) {
-      return (int)csbiInfo.dwSize.X;
+      //return (int)csbiInfo.dwSize.X; // buffer width
```
Comment (setton) [inline src/terminals.c:124]: Please remove the commented-out line. (Same comment on line 148)

File: src/terminals.c [hunk 2]
```diff
@@ -121,7 +121,9 @@ int gnatcoll_terminal_width(int forStderr) {
       GetStdHandle (forStderr ? STD_ERROR_HANDLE : STD_OUTPUT_HANDLE);
    CONSOLE_SCREEN_BUFFER_INFO csbiInfo;
    if (GetConsoleScreenBufferInfo (handle, &csbiInfo)) {
-      return (int)csbiInfo.dwSize.X;
+      //return (int)csbiInfo.dwSize.X; // buffer width 
+      return (int)(csbiInfo.srWindow.Right
+						-csbiInfo.srWindow.Left + 1); // window width
```
Comment (setton) [inline src/terminals.c:126]: The indentation of this line is off - please remove the tabs (same comment for line 150)

File: src/gnatcoll-terminal.ads [hunk 3]
```diff
@@ -180,4 +185,6 @@ private
       --  Whether the associated terminal is stdout (windows only)
    end record;
 
+
```
Comment (setton) [inline src/gnatcoll-terminal.ads:188]: Please remove these extra lines.

File: src/gnatcoll-terminal.ads [hunk 4]
```diff
@@ -140,6 +140,11 @@ package GNATCOLL.Terminal is
    --  Return the width of the terminal, or -1 if that width is either
    --  unknown or does not apply (as is the case for files for instance).
 
+   function Get_Lines (Self : Terminal_Info) return Integer;
+   --  Return the height of the terminal, or -1 if that height is either
+   --  unknown or does not apply (as is the case for files for instance).
+
+
```
Comment (setton) [inline src/gnatcoll-terminal.ads:147]: Please remove one of these blank lines.

File: src/gnatcoll-terminal.adb [hunk 5]
```diff
@@ -432,4 +432,24 @@ package body GNATCOLL.Terminal is
          return Internal (Boolean'Pos (Self.FD = Stderr));
       end if;
    end Get_Width;
+
```
Comment (setton) [inline src/gnatcoll-terminal.adb:435]: Please remove the extra blank lines - there should be at most one blank line as separation. (Same goes for lines 452-453.)

**Review by setton — CHANGES_REQUESTED** [review]
Thank you very much for this contribution. I have left a few comments - only stylistic, the code itself looks good to me. Let us know if you're happy to make those changes.

**Review by t-14 — COMMENTED** [review]
As a general comment, can you also add a new test exercising the additional code?

**Comment by fastrgv** [comment]
Ok, I think all suggested changes have been made. And I attached a test driver in this message.

[testwin_adb.txt](https://github.com/AdaCore/gnatcoll-core/files/13054161/testwin_adb.txt)

**Comment by t-14** [comment]
> And I attached a test driver in this message.

Thanks, but this needs to be added as a new test case in `testsuite/tests/terminal`.

**Comment by fastrgv** [comment]
Ok. I just added it to ~testsuite/tests/terminal/testlines.adb

**Comment by Jicquel** [comment]
Hello @fastrgv, thank you for your contribution! Your commits have been squashed and merged here: https://github.com/AdaCore/gnatcoll-core/commit/724c8dba486c913075858e21074223940795f22f.

**Pull request closed without merging**